File: src/lessons/welcome/bat/bool1/Max1020Entity.java
package lessons.welcome.bat.bool1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class Max1020Entity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(max1020((Integer)param[0], (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  int max1020(int a, int b)
  {
    /* BEGIN SOLUTION */
    int A = a > b ? a : b;
    int B = a > b ? b : a;
    if (A < 21 && A > 9)
      return A;
    if (B < 21 && B > 9)
      return B;
    return 0;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool1/Max1020Entity.scala
package lessons.welcome.bat.bool1

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class Max1020Entity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(max1020(param(0).asInstanceOf[Int], param(1).asInstanceOf[Int]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def max1020(a:Int, b:Int):Int = {
	/* BEGIN SOLUTION */
	val A = Math.max(a,b)
	val B = Math.min(a,b)
	if (A<21 && A>9)
		return A
	if (B<21 && B>9)
		return B
	return 0
	/* END SOLUTION */
	}
	/* END TEMPLATE */
}
